### PR TITLE
feat(components): [segmented] new component

### PR DESCRIPTION
**Pull request by kooriookami**
[Demo](https://preview-16258-element-plus.surge.sh/en-US/component/segmented.html)

- [x] Add Icon

### CODE REVIEW

File: docs/en-US/component/segmented.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+:::
+
+## API
+
+### Attributes
+
+| Name                  | Description                        | Type                                           | Default |
+|-----------------------|------------------------------------|------------------------------------------------|---------|
+| model-value / v-model | binding value                      | ^[string] / ^[number]                          | —       |
+| options               | data of the options                | ^[array]                                       | —       |
```
[the remark this answers is not in the document]
Comment (kooriookami) [reply docs/en-US/component/segmented.md:67]: Any examples can be refered to?

[the remark this answers is not in the document]
Comment (btea) [reply docs/en-US/component/segmented.md:67]:
```suggestion
| options               | data of the options                | ^[object]`string[] | number[] | { label: string, value: string | number, disabled?: boolean, [key: string]: any}[]`                                       | —       |
```

**Review by tolking — COMMENTED** [review]
Seems to have forgotten to add the component to the type file

typings –> components.d.ts

File: packages/components/segmented/src/segmented.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+   */
+  options: {
+    type: definePropType<Option[]>(Array),
+    default: () => [],
+  },
+  /**
+   * @description binding value
+   */
+  modelValue: {
+    type: [String, Number],
```
[the remark this answers is not in the document]
Comment (kooriookami) [reply packages/components/segmented/src/segmented.ts:26]: OK~

File: docs/.vitepress/crowdin/en-US/pages/component.json [hunk 3]
```diff
@@ -246,6 +246,11 @@
         "link": "/statistic",
         "text": "Statistic",
         "promotion": "2.2.30"
+      },
+      {
+        "link": "/segmented",
+        "text": "Segmented",
+        "promotion": "2.7.0"
```
[the remark this answers is not in the document]
Comment (kooriookami) [reply docs/.vitepress/crowdin/en-US/pages/component.json:253]: I referenced other component libraries and it's not a big deal

File: packages/components/segmented/src/types.ts [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+  | {
+      label: string
+      value: string | number | boolean
+      disabled?: boolean
+      [key: string]: any
+    }
+  | string
+  | number
+  | boolean
+  | undefined
```
Comment (btea) [inline packages/components/segmented/src/types.ts:11]: `Option` Support` undefiend`? What situation needs this support? 🤔

Comment (kooriookami) [reply packages/components/segmented/src/types.ts:11]: In order for ts to compile successfully, you can remove undefined and try building it locally

File: packages/components/segmented/src/segmented.vue [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+      >
+        <input
+          :class="ns.e('item-input')"
+          type="radio"
+          :name="name"
+          :disabled="getDisabled(item)"
+          :checked="getSelected(item)"
+          @change="handleChange(item)"
+        />
+        <div :class="ns.e('item-label')">
```
[the remark this answers is not in the document]
Comment (kooriookami) [reply packages/components/segmented/src/segmented.vue:25]: I don't think title is necessary, user may use slot, they can use tooltip if necessary.

[the remark this answers is not in the document]
Comment (btea) [reply packages/components/segmented/src/segmented.vue:25]: It makes sense, it is better for users to set it by themselves.

**Review by btea — APPROVED** [review]
👍

**Pull request merged**